fabric-data-attachment-api-v1/src/main/java/net/fabricmc/fabric/api/attachment/v1/AttachmentContainer.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.attachment.v1;
+
+import java.util.function.Supplier;
+
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.Nullable;
+
+import net.minecraft.nbt.NbtCompound;
+
+import net.fabricmc.fabric.impl.attachment.AttachmentContainerImpl;
+
+/**
+ * Maps {@linkplain AttachmentType attachment types} to attached values.
+ */
+@ApiStatus.NonExtendable
+public interface AttachmentContainer {
```

Comment:
Wish we can just use sealed, but javac doesn't let us unless we use named modules (jvm lets us though)

Reply:
Would conflict with the usual api/impl split of Fabric anyway. :wink: